apps/analysis_framework/tests/test_apis.py

```diff
@@ -155,6 +155,26 @@ def test_get_memberships(self):
         assert 'member_details' in data[0]
         assert data[0]['framework'] == framework.id
 
+    def test_get_more_memberships_data(self):
+        user1 = self.create_user()
+        user2 = self.create_user()
+        framework = self.create(AnalysisFramework)
+        framework.add_member(
+            user=user1,
+            role=framework.get_or_create_owner_role(),
+            added_by=user2
+        )
+
+        url = f'/api/v1/analysis-frameworks/{framework.id}/memberships/'
+
+        self.authenticate()
+        response = self.client.get(url)
+        self.assert_200(response)
+        data = response.data['results']
+        assert 'added_by_details' in data[0]
+        self.assertEqual(data[0]['added_by_details']['id'], user2.id)
+        assert 'role_details' in data[0]
+
```

Comment:
Let's add a test for pagination here as well (since we are handling pagination with a custom code block). Let's try with `?limit=2` with 4-5 memberships.

Reply:
Done!!